server/py/services/api/crud/alert_activation.py

```diff
@@ -13,56 +13,37 @@
 # limitations under the License.
 #
 
-from datetime import datetime, timedelta
+
+from sqlalchemy.orm import Session
 
 import mlrun.common.schemas.alert
+import mlrun.common.schemas.partition
 import mlrun.utils.singleton
+import services.api.utils.db.partitioner
 
 
 class AlertActivation(
     metaclass=mlrun.utils.singleton.Singleton,
 ):
-    @staticmethod
-    def get_partition_info(
-        partition_interval: str,
-        partition_datetime: datetime,
-    ) -> tuple[str, str, str]:
+    def create_and_drop_partitions(
+        self,
+        session: Session,
+        retention_days: int,
+    ) -> None:
```

Comment:
I now think that this wrapper function is not really useful, and we can just call the Partitioner from `main` with the `alert_activations` table name.